tests/unit/test_search_model.py

```diff
@@ -1558,3 +1569,29 @@ def test_with_active_term():
     assert isinstance(term4, Term) is True
     assert term4.field == "__glossary"
     assert term4.value == GLOSSARY_QUALIFIED_NAME
+
+
+def test_dsl_serialization_and_deserialization():
+
+    dsl1 = DSL(
+        from_=0,
+        aggregations={"test1": {"test2": {"field": "__test_field"}}},
+        size=300,
+        sort=[SortItem(field="created", order=SortOrder.ASCENDING, nested_path="test")],
+        query=Term(field="__typeName.keyword", value="Schema"),
+        track_total_hits=False,
+    )
```

Comment:
Let's also ensure we test this on complex data structures in unit tests, e.g., nested `aggregations`, multiple `sort` items in a list, and multiple queries of different types (`term`, `bool`, `range`, etc).

```suggestion
   dsl_through_model = DSL(
        from_=0,
        aggregations={"test1": {"test2": {"field": "__test_field"}}},
        size=300,
        sort=[SortItem(field="created", order=SortOrder.ASCENDING, nested_path="test")],
        query=Term(field="__typeName.keyword", value="Schema"),
        track_total_hits=False,
    )
```

Reply:
Done